### PR TITLE
[Refactor] Clean up public api and unused error codes

### PR DESCRIPTION
**Pull request by jimchou-dev**
## Purpose

* Clean up public api make Events internal and errorCode internal
* Clean up unused error codes (some error code should be event code, but refactor for it will happen in future)
* New API view for chat public preview https://apiview.dev/Assemblies/Review/94effc0dc5834079836603972a352a27

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

## How to Test

* Open...
* Click on...

## What to Check

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/ChatDemoView.swift [hunk 1]
```diff
@@ -185,14 +185,14 @@ extension ChatDemoView {
...
-            communicationIdentifier: communicationIdentifier,
+            identifier: communicationIdentifier,
             credential: communicationTokenCredential,
             endpoint: envConfigSubject.endpointUrl,
             displayName: envConfigSubject.displayName)
         guard let chatAdapter = self.chatAdapter else {
             return
         }
-        chatAdapter.events.onError = showError
+//        chatAdapter.events.onError = showError
```
Comment (emlynmac) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/ChatDemoView.swift:195]: We should remove commented code

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/ChatDemoViewController.swift [hunk 2]
```diff
@@ -204,13 +204,13 @@ class ChatDemoViewController: UIViewController {
...
     private func showError(for errorCode: String) {
-        var errorMessage = ""
-        switch errorCode {
-        case ChatCompositeErrorCode.tokenExpired:
-            errorMessage = "Token is invalid"
-        default:
-            errorMessage = "Unknown error"
-        }
+        var errorMessage = errorCode
+//        switch errorCode {
```
Comment (emlynmac) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/ChatDemoViewController.swift:208]: Here too

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatCompositeError.swift [hunk 3]
```diff
@@ -6,35 +6,24 @@
 import UIKit
 
 /// Chat Composite runtime error types.
-public struct ChatCompositeErrorCode {
-    /// Error when local user fails to join a chat.
-    public static let chatConnect: String = "chatConnect"
+struct ChatCompositeErrorCode {
+    /// Error when adapter fails to connect to a chat.
+    static let chatConnect: String = "chatConnect"
```
Comment (emlynmac) [inline AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatCompositeError.swift:11]: Maybe name this error `connectFailed` ?

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatCompositeError.swift [hunk 4]
```diff
@@ -6,35 +6,24 @@
...
 
-    /// Error when a chat disconnects unexpectedly or fails on ending.
-    public static let chatEnd: String = "chatEnd"
+    /// Error when the input token is not authorized for the threadId.
+    static let chatUnauthorized: String = "chatUnauthorized"
```
Comment (emlynmac) [inline AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatCompositeError.swift:14]: maybe `authorizationFailed` ?

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatCompositeError.swift [hunk 5]
```diff
@@ -6,35 +6,24 @@
...
 
-    /// Error when the input token is expired.
-    public static let tokenExpired: String = "tokenExpired"
+    /// Error when a chat disconnects unexpectedly or fails on ending. (event code?)
+    static let chatEnd: String = "chatEnd"
```
Comment (emlynmac) [inline AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatCompositeError.swift:17]: Maybe `disconnectFailed`?

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatCompositeError.swift [hunk 6]
```diff
@@ -6,35 +6,24 @@
...
     static let chatDenied: String = "chatDenied"
 
-    /// Error when local user fails to hold a chat.
-    static let chatHold: String = "chatHold"
-
-    /// Error when local user fails to resume a chat.
-    static let chatResume: String = "chatResume"
-
     /// Error when local user fails to send message.
     static let sendMessage: String = "sendMessage"
```
Comment (emlynmac) [inline AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatCompositeError.swift:26]: `messageSendFailed` perhaps?

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatCompositeError.swift [hunk 7]
```diff
@@ -6,35 +6,24 @@
...
-    /// Error when the input token is expired.
-    public static let tokenExpired: String = "tokenExpired"
+    /// Error when a chat disconnects unexpectedly or fails on ending. (event code?)
+    static let chatEnd: String = "chatEnd"
 
-    public static let showComposite: String = "showComposite"
-
-    /// Error when a participant is evicted from the chat by another participant
+    /// Error when a participant is evicted from the chat by another participant (event code?)
     static let chatEvicted: String = "chatEvicted"
```
Comment (emlynmac) [inline AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatCompositeError.swift:20]: Is this really an error case or actually part of the regular cases?

Comment (jimchou-dev) [reply AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatCompositeError.swift:20]: They're not really an error code, move them to ChatCompositeEventCode (but likely will be refactored in the future when we implement eventHandler)

**Review by JoshuaLai — APPROVED** [review]

**Review by vhuseinova-msft — APPROVED** [review]

**Pull request merged**